### DISCUSSION TITLE
Would a PR to add config elements and modify sections to be responsive by default be of interest?

### DISCUSSION THREAD

**Discussion by roni-estein** [filamentphp/filament#693] [General]
I mostly program for a mobile target but the usage stats tell an interesting story. The staff who use this most, are at desks using desktop or laptop large machines. Or walking with larger tablets. To accommodate this, I've made all my "sections" a bit different. I've added changes as traits which defer to config options you should publish in your app that can be overridden at the class or method level as well.

1. Make Classes responsive by default. Nothing too fancy, you can publish your own views and make them whatever you want. But a Basic TailwindCSS version of responsive.
2. Remove the border and padding from sections by default and add them in a containing element.
3. add a divider over the section by default just a UI nicety,

Here is a Quick example

https://user-images.githubusercontent.com/8517475/140798981-bc576862-c7cd-47f2-a82b-18d22f2dcc09.mov

**Comment by ralphjsmit** [comment]
Hey Roni, I just came across this and I noticed that you're pulling in a lot of traits with rather interesting names. Do you mind sharing how some of them work, like CanHaveTopLine? Do they integrate with Filament are those not Filament-related?

**Reply by roni-estein** [reply]
That's a fair point. Also I'm doing this on a phone heading to a meeting, but I wanted to give you a quick response.

I've tried to keep the coding style that existed already. In places where I would have reproduced that same code sections / forms & bounding elements, I try to pull in a behaviour as a trait. (I do also use single use traits from time to time)

Many times it's just dealing with a simple attribute that you might reach for like autocapitalization, another trait I PR'd that could you use extraAttributes. But in the end it's just ux that takes filament to the next level. 

Another goal of mine is to eventually make it sensible for most users to not even need a blade file to go with a form component while keeping the form readable and clean and the view very predictable.

Lastly, this is just syntactic sugar. You could accomplish this in lots of different ways and in many easier fashions. Lots of apps that I work on have hundreds of forms. The idea of not needing "most" of the blade files is basically awesome 😎. If you are working on an app with 10's of forms this might be nice but you really wouldn't care.

Hope that's helpful 😀

**Reply by ralphjsmit** [reply]
That's certainly helpful! Yeah, so those traits just provide easy functions that you include in the `getFormSchema()` function?

Well, even though an app doesn't contain tens of forms, it feels way better to use traits use reusability and prevent duplication.

**Comment by danharrin** [comment]
Hey! That section looks great. You could release a plugin for this! I love the idea of more layout components but many are dependent on the styling of the app, so a separate package for this could be useful for loads of Tailwind UI users.

Maybe `composer require roni-estein/filament-responsive-section-plugin`? 😛

**Reply by roni-estein** [reply]
Totally 😀

**Reply by roni-estein** [reply]
Is there a good example of a plug-in I can work off of? I haven't searched yet 👀

**Reply by roni-estein** [reply]
The funny thing is that isn't even tailwindUI! Although I should probably use it more for all the goodies it adds :) Its from my jillions of blade packages of components prior to tailwind releasing UI. Most use tailwind css but not all. I should update everything to tailwind latest at some point and release them over time, but I don't want to get into any licensing issues.

**Reply by ralphjsmit** [reply]
No? It's not Tailwind🤯 It looks so similar

**Reply by danharrin** [reply]
I can put an example plugin together, but honestly it's just a view file in `/views` and a component file in `/src` haha - you don't need to register anything Filamenty for `filament/forms` plugins, [only the views directory](https://laravel.com/docs/8.x/packages#views).
